unit-tests/Extension/Oo/Scopes/PrivateScopeTest.php

```diff
@@ -13,12 +13,175 @@
+    public function testShouldNotSetPrivatePropertyNewInternal()
+    {
+        $tester = new PropertyTester();
+        $tester->setPropertyNew(TestScopeExtending::class, 'privateProperty', 'test');
+        $this->assertFalse(true, 'This should not be called');
+    }
+
+    public function testShouldSetPrivatePropertyObjPhp()
+    {
+        $object = new TestScopePhpMagic();
+        $tester = new PropertyTester();
+        $this->assertEquals('test', $tester->setPropertyObj($object, 'privateProperty', 'test'));
+        $this->assertEquals(1, $object->setCount);
+    }
+
+    public function testShouldSetPrivatePropertyNewPhp()
+    {
+        $tester = new PropertyTester();
+        $obj = $tester->setPropertyNew(TestScopePhpMagic::class, 'privateProperty', 'test');
+        $this->assertEquals('test', $obj->privateProperty);
+        $this->assertEquals(1, $obj->setCount);
+    }
+
+    public function testShouldSetPrivatePropertyObjInternal()
+    {
+        $object = new TestScopeExtendingMagic();
+        $tester = new PropertyTester();
+        $this->assertEquals('test', $tester->setPropertyObj($object, 'privateProperty', 'test'));
+        $this->assertEquals(1, $object->setCount);
+    }
+
+    public function testShouldSetPrivatePropertyNewInternal()
+    {
+        $tester = new PropertyTester();
+        $obj = $tester->setPropertyNew(TestScopeExtendingMagic::class, 'privateProperty', 'test');
+        $this->assertEquals('test', $obj->privateProperty);
+        $this->assertEquals(1, $obj->setCount);
+    }
+
+    /**
+     * @expectedException \Error
+     */
+    public function testShouldNotSetPrivatePropertyViaThis()
+    {
+        $obj = new TestScopeExtending();
+        $obj->setProperty('privateProperty', 'test');
+        $this->assertFalse(true, 'This should not be called');
+    }
+
+    /**
+     * @expectedException \Error
+     */
+    public function testShouldNotSetPrivatePropertyExtendedMagicObjInternal()
+    {
+        $obj = new TestScopeExtendingMagic();
+        $tester = new PropertyTester();
+        $this->assertEquals('test', $tester->setPropertyObj($obj, 'privateProperty2', 'test'));
+        $this->assertFalse(true, 'This should not be called');
+    }
+
+    /**
+     * @expectedException \Error
+     */
+    public function testShouldNotSetPrivatePropertyExtendedMagicNewInternal()
+    {
+        $tester = new PropertyTester();
+        $tester->setPropertyNew(TestScopeExtendingMagic::class, 'privateProperty2', 'test');
+        $this->assertFalse(true, 'This should not be called');
+    }
+
+    /**
+     * @expectedException \Error
+     */
+    public function testShouldNotSetPrivatePropertyExtendedMagicObjPhp()
+    {
+        $obj = new TestScopePhpMagicExtending();
+        $tester = new PropertyTester();
+        $tester->setPropertyObj($obj, 'privateProperty2', 'test');
+        $this->assertFalse(true, 'This should not be called');
+    }
+
+    /**
+     * @expectedException \Error
```

Comment:
For exceptions test - better is to use [built-in phpunit assertion](https://phpunit.readthedocs.io/en/latest/writing-tests-for-phpunit.html#testing-exceptions) `$this->expectedException()`